docker-compose.storage.minio.yml

```diff
@@ -16,12 +16,18 @@ services:
         environment:
             MINIO_HTTP_TRACE: /dev/stdout
         command: server /export
+        healthcheck:
+            test: ["CMD", "curl", "-f", "http://localhost:9000/minio/health/live"]
+            interval: 1s
```

Comment:
I actually have played with that a little bit (see pushes in https://github.com/mendersoftware/integration/pull/2053 for details) and this combination of interval, timeout and retries looked stable. But if to be honest it's hard for me to understand where 30s comes from...

Reply:
I tested it locally and the best result is with `interval: 1s` and `start_period: 1s`. In this case locally it starts in the rage between 1.5s and 4s. How it works is it waits for `start_period` and then executes the check and repeats it with the `interval`. With `start_period: 5s` executed on not loaded server it will no difference between `interval: 10s` and `interval: 1s`, because very first check after `start_period` will pass, but on a loaded server with `interval: 10s` and failed 1st check it will not start faster than 15s. To me it looks like `start_period: 5s` + `interval: 1s` is good combination because locally it's only 1-2 secs slower that it can be in the best case, but in the same time in most cases very first check will pass and it won't generate any additional load (if executing HTTP GET by curl can be called so) even with `interval: 1s`. Not really sure what we win with 10s interval.

Reply:
ah ok 👌 for me this was mainly about the recurring GET calls (where I think the recommendation was to keep the interval higher for less critical services to keep the load on the service down - but I guess that depends on the endpoint that's called to determine container health) - if that's not an issue at all 🚀 merge ahead!

Reply:
To make it in the best way I believe it would be nice to separate those settings for local runs/mender servers and our tests. For local runs it could be something like: `start_period: 10s` and `interval: 60s`, but for the tests the fastest feedback we get is better.

Reply:
I think we should not put this in `docker-compose.storage.minio.yml`, which may end up in production configuration for our users. Instead we should add it to `docker-compose.demo.yml` [here](https://github.com/mendersoftware/integration/blob/b3a0ad7e219bcf9d16896b9ceb5fa7baf4bf5553/docker-compose.demo.yml#L69), which will affect users only when running demo setup.

For testing, we use `docker-compose.demo.yml` with some [modifications](https://github.com/mendersoftware/integration/blob/b3a0ad7e219bcf9d16896b9ceb5fa7baf4bf5553/tests/run.sh#L89). Injecting there the `healthcheck` is very ugly.

Reply:
In general healthchecks are even more important for production systems. I don't mind to move it to `docker-compose.demo.yml` and leave `docker-compose.storage.minio.yml` without healthcheck to make some beauty (even considering that it's subjective judgment 🙃), considering that docker-compose based installation is unlikely can be called a real production one.

Reply:
> In general healthchecks are even more important for production systems. (...)

Good point. We could then have it in the base file with conservative intervals and override only the intervals to more aggressive ones for the `demo` setup.